### PR TITLE
store/tikv: resolve ReplicaReadType dependencies

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
tikv's `ReplicaReadType` is used everywhere. We want the dependency between tidb and client go to be as simple as possible.

Part of https://github.com/pingcap/tidb/issues/24302

### What is changed and how it works?
What's Changed:
- Define new ReplicaReadType in /kv
- Replace `tikv.ReplicaReadType` with `kv.ReplicaReadType`.
- Map the 2 `ReplicaReadType`s in driver layer.

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/kv/store_vars.go [hunk 1]
```diff
@@ -25,15 +25,14 @@ type ReplicaReadType byte
 
 const (
 	// ReplicaReadLeader stands for 'read from leader'.
-	ReplicaReadLeader ReplicaReadType = 1 << iota
+	ReplicaReadLeader ReplicaReadType = iota
```
Comment (tisonkun) [inline store/tikv/kv/store_vars.go:28]: We treat both `0` and `1` previously as `ReplicaReadLeader` so for `ReplicaReadLeader` it is ok. But now `ReplicaReadFollower` becomes `1` which might cause problem.

Did you check the usage of `ReplicaReadType`? Are they all from `IsFollowerRead`? If so, then it is ok to do this change.

**Review by tisonkun — COMMENTED** [review]
Comment inline. Rest LGTM.

**Review by AndreMouche — DISMISSED** [review]
LGTM

File: store/tikv/kv/store_vars.go (hunk 1 above)
Comment (AndreMouche) [reply store/tikv/kv/store_vars.go:28]: It already causes CI to fail, please fix it.

Comment (disksing) [reply store/tikv/kv/store_vars.go:28]: fixed. It causes problem if we use `0x01` instead of `ReplicaReadLeader`.

**Review by AndreMouche — APPROVED** [review]
LGTM

**Comment by disksing** [comment]
filed an issue for the CI issue https://github.com/pingcap/tidb/issues/24679
starting re-run.

**Pull request merged**